pkg/plugins/discovery.go

```diff
@@ -103,3 +94,10 @@ func readPluginJSONInfo(name, path string) (*Plugin, error) {
 
 	return &p, nil
 }
+
+func pluginPaths(base, name, ext string) []string {
+	return []string{
+		filepath.Join(base, name+ext),
```

Comment:
I think we're fine with making plugin.sock win for now. We can revisit if it causes too much confusion or creates too many issues.